compose/ui/ui-text/src/desktopMain/kotlin/androidx/compose/ui/text/platform/DesktopFont.desktop.kt

```diff
@@ -47,7 +48,8 @@ actual sealed class PlatformFont : Font {
 class ResourceFont internal constructor(
     val name: String,
     override val weight: FontWeight = FontWeight.Normal,
-    override val style: FontStyle = FontStyle.Normal
+    override val style: FontStyle = FontStyle.Normal,
+    override val variationSettings: FontVariation.Settings = FontVariation.Settings(weight, style),
```

Comment:
If variationSettings is set, it will follow the settings of variationSettings.

Reply:
In summary:

1. Non-variable fonts define the font style only according to weight and style.

2. Variable fonts define the font style only according to variationSettings.

3. Because the default values of variationSettings are derived from style and style, in the case of variable fonts with weight, style, and variationSettings set, the font style will be determined by variationSettings.